diff --git a/src/main/java/finalforeach/cosmicreach/worldgen/generators/InfiniteBasicZoneGenerator.java b/src/main/java/finalforeach/cosmicreach/worldgen/generators/InfiniteBasicZoneGenerator.java
index 985c516..b0c47fa 100644
--- a/src/main/java/finalforeach/cosmicreach/worldgen/generators/InfiniteBasicZoneGenerator.java
+++ b/src/main/java/finalforeach/cosmicreach/worldgen/generators/InfiniteBasicZoneGenerator.java
@@ -14,6 +14,7 @@ import finalforeach.cosmicreach.worldgen.ZoneGenerator;
 import finalforeach.cosmicreach.worldgen.noise.SimplexNoise;
 import finalforeach.cosmicreach.worldgen.noise.WhiteNoise;
 
+
 public class InfiniteBasicZoneGenerator extends ZoneGenerator {
    private SimplexNoise simplexNoise;
    private SimplexNoise simplexNoise2;
@@ -33,12 +34,7 @@ public class InfiniteBasicZoneGenerator extends ZoneGenerator {
    BlockState sandBlock = Block.SAND.getDefaultBlockState();
    BlockState oakLeavesBlock = Block.SNOW.getDefaultBlockState();
    BlockState oakLogBlock = Block.TREELOG.getDefaultBlockState();
-   IBlockDataFactory<BlockState> layeredChunkDataFactory = new IBlockDataFactory<BlockState>() {
-      @Override
-      public IBlockData<BlockState> createChunkData() {
-         return new SingleBlockData<>(InfiniteBasicZoneGenerator.this.airblock);
-      }
-   };
+   IBlockDataFactory<BlockState> layeredChunkDataFactory = () -> new SingleBlockData<>(InfiniteBasicZoneGenerator.this.airblock);
 
    @Override
    public String getSaveKey() {
