diff --git a/src/main/java/finalforeach/cosmicreach/worldgen/generators/MoonZoneGenerator.java b/src/main/java/finalforeach/cosmicreach/worldgen/generators/MoonZoneGenerator.java
index 879fc04..e05217c 100644
--- a/src/main/java/finalforeach/cosmicreach/worldgen/generators/MoonZoneGenerator.java
+++ b/src/main/java/finalforeach/cosmicreach/worldgen/generators/MoonZoneGenerator.java
@@ -27,12 +27,7 @@ public class MoonZoneGenerator extends ZoneGenerator {
    private WhiteNoise whiteNoiseCratersZ;
    private WhiteNoise whiteNoiseCratersH;
    private WhiteNoise whiteNoiseCratersV;
-   IBlockDataFactory<BlockState> layeredChunkDataFactory = new IBlockDataFactory<BlockState>() {
-      @Override
-      public IBlockData<BlockState> createChunkData() {
-         return new LayeredBlockData<>(MoonZoneGenerator.this.airblock);
-      }
-   };
+   IBlockDataFactory<BlockState> layeredChunkDataFactory = () -> new LayeredBlockData<>(MoonZoneGenerator.this.airblock);
 
    @Override
    public void create() {
