diff --git a/src/main/java/finalforeach/cosmicreach/worldgen/generators/NostalgicIslandZoneGenerator.java b/src/main/java/finalforeach/cosmicreach/worldgen/generators/NostalgicIslandZoneGenerator.java
index da6baf3..a13d1c0 100644
--- a/src/main/java/finalforeach/cosmicreach/worldgen/generators/NostalgicIslandZoneGenerator.java
+++ b/src/main/java/finalforeach/cosmicreach/worldgen/generators/NostalgicIslandZoneGenerator.java
@@ -20,12 +20,7 @@ public class NostalgicIslandZoneGenerator extends ZoneGenerator {
    BlockState sandBlock = this.getBlockStateInstance("base:sand[default]");
    BlockState dirtBlock = this.getBlockStateInstance("base:dirt[default]");
    private SimplexNoise simplexNoise;
-   IBlockDataFactory<BlockState> chunkDataFactory = new IBlockDataFactory<BlockState>() {
-      @Override
-      public IBlockData<BlockState> createChunkData() {
-         return new SingleBlockData<>(NostalgicIslandZoneGenerator.this.airblock);
-      }
-   };
+   IBlockDataFactory<BlockState> chunkDataFactory = () -> new SingleBlockData<>(NostalgicIslandZoneGenerator.this.airblock);
 
    @Override
    public void create() {
