diff --git a/src/main/java/libs/opensimplex2/OpenSimplex2.java b/src/main/java/libs/opensimplex2/OpenSimplex2.java
index 2e49af0..b37830d 100644
--- a/src/main/java/libs/opensimplex2/OpenSimplex2.java
+++ b/src/main/java/libs/opensimplex2/OpenSimplex2.java
@@ -1,5 +1,6 @@
 package libs.opensimplex2;
 
+
 public class OpenSimplex2 {
    private static final long PRIME_X = 5910200641878280303L;
    private static final long PRIME_Y = 6452764530575939509L;
@@ -317,7 +318,7 @@ public class OpenSimplex2 {
       hash *= 6026932503003350773L;
       hash ^= hash >> 58;
       int gi = (int)hash & 254;
-      return GRADIENTS_2D[gi | 0] * dx + GRADIENTS_2D[gi | 1] * dy;
+      return GRADIENTS_2D[gi] * dx + GRADIENTS_2D[gi | 1] * dy;
    }
 
    private static float grad(long seed, long xrvp, long yrvp, long zrvp, float dx, float dy, float dz) {
@@ -325,7 +326,7 @@ public class OpenSimplex2 {
       hash *= 6026932503003350773L;
       hash ^= hash >> 58;
       int gi = (int)hash & 1020;
-      return GRADIENTS_3D[gi | 0] * dx + GRADIENTS_3D[gi | 1] * dy + GRADIENTS_3D[gi | 2] * dz;
+      return GRADIENTS_3D[gi] * dx + GRADIENTS_3D[gi | 1] * dy + GRADIENTS_3D[gi | 2] * dz;
    }
 
    private static float grad(long seed, long xsvp, long ysvp, long zsvp, long wsvp, float dx, float dy, float dz, float dw) {
@@ -333,7 +334,7 @@ public class OpenSimplex2 {
       hash *= 6026932503003350773L;
       hash ^= hash >> 57;
       int gi = (int)hash & 2044;
-      return GRADIENTS_4D[gi | 0] * dx + GRADIENTS_4D[gi | 1] * dy + GRADIENTS_4D[gi | 2] * dz + GRADIENTS_4D[gi | 3] * dw;
+      return GRADIENTS_4D[gi] * dx + GRADIENTS_4D[gi | 1] * dy + GRADIENTS_4D[gi | 2] * dz + GRADIENTS_4D[gi | 3] * dw;
    }
 
    private static int fastFloor(double x) {
