diff --git a/src/main/java/libs/opensimplex2/OpenSimplex2S.java b/src/main/java/libs/opensimplex2/OpenSimplex2S.java
index b9bee81..93ae096 100644
--- a/src/main/java/libs/opensimplex2/OpenSimplex2S.java
+++ b/src/main/java/libs/opensimplex2/OpenSimplex2S.java
@@ -1,5 +1,6 @@
 package libs.opensimplex2;
 
+
 public class OpenSimplex2S {
    private static final long PRIME_X = 5910200641878280303L;
    private static final long PRIME_Y = 6452764530575939509L;
@@ -468,7 +469,7 @@ public class OpenSimplex2S {
       long ysvp = (long)ysb * 6452764530575939509L;
       long zsvp = (long)zsb * 6614699811220273867L;
       long wsvp = (long)wsb * 6254464313819354443L;
-      int index = (fastFloor(xs * 4.0) & 3) << 0 | (fastFloor(ys * 4.0) & 3) << 2 | (fastFloor(zs * 4.0) & 3) << 4 | (fastFloor(ws * 4.0) & 3) << 6;
+      int index = (fastFloor(xs * 4.0) & 3) | (fastFloor(ys * 4.0) & 3) << 2 | (fastFloor(zs * 4.0) & 3) << 4 | (fastFloor(ws * 4.0) & 3) << 6;
       float value = 0.0F;
       int secondaryIndexStartAndStop = LOOKUP_4D_A[index];
       int secondaryIndexStart = secondaryIndexStartAndStop & 65535;
@@ -496,7 +497,7 @@ public class OpenSimplex2S {
       hash *= 6026932503003350773L;
       hash ^= hash >> 58;
       int gi = (int)hash & 254;
-      return GRADIENTS_2D[gi | 0] * dx + GRADIENTS_2D[gi | 1] * dy;
+      return GRADIENTS_2D[gi] * dx + GRADIENTS_2D[gi | 1] * dy;
    }
 
    private static float grad(long seed, long xrvp, long yrvp, long zrvp, float dx, float dy, float dz) {
@@ -512,7 +513,7 @@ public class OpenSimplex2S {
       hash *= 6026932503003350773L;
       hash ^= hash >> 57;
       int gi = (int)hash & 2044;
-      return GRADIENTS_4D[gi | 0] * dx + GRADIENTS_4D[gi | 1] * dy + GRADIENTS_4D[gi | 2] * dz + GRADIENTS_4D[gi | 3] * dw;
+      return GRADIENTS_4D[gi] * dx + GRADIENTS_4D[gi | 1] * dy + GRADIENTS_4D[gi | 2] * dz + GRADIENTS_4D[gi | 3] * dw;
    }
 
    private static int fastFloor(double x) {
