diff --git a/src/main/java/finalforeach/cosmicreach/items/ItemMergeStrategy.java b/src/main/java/finalforeach/cosmicreach/items/ItemMergeStrategy.java
index 1a2b958..3d5aa79 100644
--- a/src/main/java/finalforeach/cosmicreach/items/ItemMergeStrategy.java
+++ b/src/main/java/finalforeach/cosmicreach/items/ItemMergeStrategy.java
@@ -2,5 +2,5 @@ package finalforeach.cosmicreach.items;
 
 public enum ItemMergeStrategy {
    PRESERVE,
-   ONLY_ONE_SLOT;
+   ONLY_ONE_SLOT
 }
