diff --git a/src/main/java/finalforeach/cosmicreach/rendering/blockmodels/BlockModelJson.java b/src/main/java/finalforeach/cosmicreach/rendering/blockmodels/BlockModelJson.java
index 30814f5..02f4e0b 100644
--- a/src/main/java/finalforeach/cosmicreach/rendering/blockmodels/BlockModelJson.java
+++ b/src/main/java/finalforeach/cosmicreach/rendering/blockmodels/BlockModelJson.java
@@ -44,7 +44,7 @@ public class BlockModelJson extends BlockModel {
 
       return models.get(key);
    }
-
+   @SuppressWarnings("unchecked")
    public static BlockModelJson fromJson(String modelJson, int rotXZ) {
       Json json = new Json();
       BlockModelJson b = json.fromJson(BlockModelJson.class, modelJson);
@@ -173,10 +173,10 @@ public class BlockModelJson extends BlockModel {
                   }
                   break;
                case 270:
-                  float fxa = 16.0F - boundsX1;
-                  float fxb = 16.0F - boundsX2;
-                  float fza = 16.0F - boundsZ1;
-                  float fzb = 16.0F - boundsZ2;
+                  fxa = 16.0F - boundsX1;
+                  fxb = 16.0F - boundsX2;
+                  fza = 16.0F - boundsZ1;
+                  fzb = 16.0F - boundsZ2;
                   c.localBounds[0] = Math.min(fza, fzb);
                   c.localBounds[2] = Math.min(fxa, fxb);
                   c.localBounds[3] = Math.max(fza, fzb);
