diff --git a/src/main/java/finalforeach/cosmicreach/rendering/ChunkBatch.java b/src/main/java/finalforeach/cosmicreach/rendering/ChunkBatch.java
index 79e85eb..8986653 100644
--- a/src/main/java/finalforeach/cosmicreach/rendering/ChunkBatch.java
+++ b/src/main/java/finalforeach/cosmicreach/rendering/ChunkBatch.java
@@ -215,6 +215,6 @@ public class ChunkBatch {
       }
    }
 
-   static record MeshDataMetadata(int vertexPosition, int numVertices) {
+   record MeshDataMetadata(int vertexPosition, int numVertices) {
    }
 }
