diff --git a/src/main/java/finalforeach/cosmicreach/rendering/ChunkMeshGroup.java b/src/main/java/finalforeach/cosmicreach/rendering/ChunkMeshGroup.java
index 74059cf..1f0c937 100644
--- a/src/main/java/finalforeach/cosmicreach/rendering/ChunkMeshGroup.java
+++ b/src/main/java/finalforeach/cosmicreach/rendering/ChunkMeshGroup.java
@@ -32,7 +32,7 @@ public class ChunkMeshGroup implements IChunkMeshGroup<Array<MeshData>>, Disposa
 
    private static Array<MeshData> getMeshData(ISavedChunk<?> savedChunk) {
       Chunk chunk = (Chunk)savedChunk;
-      IBlockData<BlockState> blockData = chunk.getBlockData();
+      IBlockData<BlockState> blockData = (IBlockData<BlockState>) chunk.getBlockData();
       BlockState airBlockState = Block.AIR.getDefaultBlockState();
       if (blockData.isEntirely(airBlockState)) {
          return emptyMeshDatas;
