diff --git a/src/main/java/finalforeach/cosmicreach/constants/Direction.java b/src/main/java/finalforeach/cosmicreach/constants/Direction.java
index 9bc00ee..629b904 100644
--- a/src/main/java/finalforeach/cosmicreach/constants/Direction.java
+++ b/src/main/java/finalforeach/cosmicreach/constants/Direction.java
@@ -2,6 +2,7 @@ package finalforeach.cosmicreach.constants;
 
 import com.badlogic.gdx.utils.Array;
 
+
 public enum Direction {
    NEG_X(-1, 0, 0),
    POS_X(1, 0, 0),
@@ -59,7 +60,7 @@ public enum Direction {
       return this.oppositeDirection;
    }
 
-   private Direction(int xOff, int yOff, int zOff) {
+   Direction(int xOff, int yOff, int zOff) {
       this.xOff = xOff;
       this.yOff = yOff;
       this.zOff = zOff;
