diff --git a/src/main/java/finalforeach/cosmicreach/ui/debug/DebugInfo.java b/src/main/java/finalforeach/cosmicreach/ui/debug/DebugInfo.java
index 5a8cbe0..aaf9fb6 100644
--- a/src/main/java/finalforeach/cosmicreach/ui/debug/DebugInfo.java
+++ b/src/main/java/finalforeach/cosmicreach/ui/debug/DebugInfo.java
@@ -13,6 +13,7 @@ import finalforeach.cosmicreach.ui.FontRenderer;
 import finalforeach.cosmicreach.world.Zone;
 import java.text.DecimalFormat;
 
+
 public class DebugInfo {
    private static Array<DebugItem> items = new Array<>();
    private static String entireStr;
