diff --git a/src/main/java/finalforeach/cosmicreach/ui/HorizontalAnchor.java b/src/main/java/finalforeach/cosmicreach/ui/HorizontalAnchor.java
index d06f619..9e3b84e 100644
--- a/src/main/java/finalforeach/cosmicreach/ui/HorizontalAnchor.java
+++ b/src/main/java/finalforeach/cosmicreach/ui/HorizontalAnchor.java
@@ -3,5 +3,5 @@ package finalforeach.cosmicreach.ui;
 public enum HorizontalAnchor {
    LEFT_ALIGNED,
    CENTERED,
-   RIGHT_ALIGNED;
+   RIGHT_ALIGNED
 }
