diff --git a/src/main/java/finalforeach/cosmicreach/GameAssetLoader.java b/src/main/java/finalforeach/cosmicreach/GameAssetLoader.java
index a1e941e..ec9c3e6 100644
--- a/src/main/java/finalforeach/cosmicreach/GameAssetLoader.java
+++ b/src/main/java/finalforeach/cosmicreach/GameAssetLoader.java
@@ -1,7 +1,6 @@
 package finalforeach.cosmicreach;
 
 import com.badlogic.gdx.Gdx;
-import com.badlogic.gdx.assets.AssetManager;
 import com.badlogic.gdx.audio.Sound;
 import com.badlogic.gdx.files.FileHandle;
 import finalforeach.cosmicreach.io.SaveLocation;
@@ -10,7 +9,6 @@ import java.util.HashMap;
 public class GameAssetLoader {
    public static final HashMap<String, FileHandle> ALL_ASSETS = new HashMap<>();
    public static final HashMap<String, Sound> ALL_SOUNDS = new HashMap<>();
-   public static AssetManager assetManager = new AssetManager();
 
    public static FileHandle loadAsset(String fileName) {
       return loadAsset(fileName, false);
