diff --git a/src/main/java/finalforeach/cosmicreach/WorldLoader.java b/src/main/java/finalforeach/cosmicreach/WorldLoader.java
index 244f2d4..7936b8f 100644
--- a/src/main/java/finalforeach/cosmicreach/WorldLoader.java
+++ b/src/main/java/finalforeach/cosmicreach/WorldLoader.java
@@ -204,8 +204,7 @@ public class WorldLoader {
                         return (int)(dBSq - dASq);
                      }
                   );
-            } catch (Exception var23) {
-            }
+            } catch (Exception ignored) {}
 
             while(this.chunkColumnsToGenerate.notEmpty() && !this.requestSave) {
                ChunkColumn col = this.chunkColumnsToGenerate.pop();
