diff --git a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java
index 2f7ded0..fef4b3d 100644
--- a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java
+++ b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java
@@ -3,6 +3,7 @@ package finalforeach.cosmicreach.savelib.blockdata.layers;
 import finalforeach.cosmicreach.savelib.blockdata.LayeredBlockData;
 import java.util.HashMap;
 
+@SuppressWarnings("unchecked")
 public class SharedBlockSingleLayer<T> extends BlockSingleLayer<T> {
    private static HashMap<Object, SharedBlockSingleLayer<?>> sharedInstances = new HashMap<>();
 
@@ -29,6 +30,6 @@ public class SharedBlockSingleLayer<T> extends BlockSingleLayer<T> {
          }
       }
 
-      return shared;
+      return (SharedBlockSingleLayer<T>) shared;
    }
 }
