diff --git a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java
index 2f7ded0..1fbcb81 100644
--- a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java
+++ b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/layers/SharedBlockSingleLayer.java
@@ -29,6 +29,6 @@ public class SharedBlockSingleLayer<T> extends BlockSingleLayer<T> {
          }
       }
 
-      return shared;
+      return (SharedBlockSingleLayer<T>) shared;
    }
 }
