diff --git a/src/main/java/finalforeach/cosmicreach/rendering/meshes/IntIndexData.java b/src/main/java/finalforeach/cosmicreach/rendering/meshes/IntIndexData.java
index 3424f15..fe6c929 100644
--- a/src/main/java/finalforeach/cosmicreach/rendering/meshes/IntIndexData.java
+++ b/src/main/java/finalforeach/cosmicreach/rendering/meshes/IntIndexData.java
@@ -78,30 +78,4 @@ public class IntIndexData implements Disposable {
          this.isBound = false;
       }
    }
-
-   public static IntIndexData createQuadIndexData(int numIndices) {
-      numIndices = (int)Math.ceil((double)((float)numIndices / 6.0F)) * 6;
-      int[] indices = createQuadIndices(numIndices);
-      IntIndexData indexData = new IntIndexData(true, numIndices);
-      indexData.setIndices(indices);
-      return indexData;
-   }
-
-   public static int[] createQuadIndices(int numIndices) {
-      numIndices = (int)Math.ceil((double)((float)numIndices / 6.0F)) * 6;
-      int[] indices = new int[numIndices];
-      int f = 0;
-
-      for(int i = 0; i < numIndices; i += 6) {
-         indices[i] = f + 0;
-         indices[i + 1] = f + 1;
-         indices[i + 2] = f + 2;
-         indices[i + 3] = f + 2;
-         indices[i + 4] = f + 3;
-         indices[i + 5] = f + 0;
-         f += 4;
-      }
-
-      return indices;
-   }
 }
