diff --git a/src/main/java/finalforeach/cosmicreach/blocks/BlockState.java b/src/main/java/finalforeach/cosmicreach/blocks/BlockState.java
index ed325c8..de2c21b 100644
--- a/src/main/java/finalforeach/cosmicreach/blocks/BlockState.java
+++ b/src/main/java/finalforeach/cosmicreach/blocks/BlockState.java
@@ -16,6 +16,7 @@ import java.util.HashSet;
 import java.util.Map;
 import java.util.Map.Entry;
 
+
 public class BlockState implements IBlockState {
    public transient ItemBlock item;
    private transient Block block;
@@ -272,7 +273,7 @@ public class BlockState implements IBlockState {
    public static String getModifiedSaveKey(String oldSaveKey, String paramName, String paramVal) {
       String newSaveKey = oldSaveKey;
       String paramSet = paramName + "=" + paramVal;
-      if (paramVal.length() == 0) {
+      if (paramVal.isEmpty()) {
          paramSet = "";
       }
 
