diff --git a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/BlockDataCompactor.java b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/BlockDataCompactor.java
index e9aefee..08ffd05 100644
--- a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/BlockDataCompactor.java
+++ b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/BlockDataCompactor.java
@@ -6,7 +6,7 @@ import finalforeach.cosmicreach.savelib.blockdata.layers.SharedBlockSingleLayer;
 
 public class BlockDataCompactor {
    public static <T> IBlockData<T> compact(IBlockData<T> blockData) {
-      if (blockData instanceof LayeredBlockData layered) {
+      if (blockData instanceof LayeredBlockData<T> layered) {
          IBlockLayer<T>[] allLayers = layered.getLayers();
 
          for(int yLevel = 0; yLevel < allLayers.length; ++yLevel) {
