diff --git a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/LayeredBlockData.java b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/LayeredBlockData.java
index aa706b6..cda3078 100644
--- a/src/main/java/finalforeach/cosmicreach/savelib/blockdata/LayeredBlockData.java
+++ b/src/main/java/finalforeach/cosmicreach/savelib/blockdata/LayeredBlockData.java
@@ -124,8 +124,8 @@ public class LayeredBlockData<T> implements IBlockData<T> {
    }
 
    public void setLayer(int yLevel, IBlockLayer<T> layer) {
-      if (layer instanceof SharedBlockSingleLayer shared && !this.paletteHasValue(shared.blockValue)) {
-         this.addToPalette(shared.blockValue);
+      if (layer instanceof SharedBlockSingleLayer shared && !this.paletteHasValue((T) shared.blockValue)) {
+         this.addToPalette((T) shared.blockValue);
       }
 
       this.layers[yLevel] = layer;
