diff --git a/src/main/java/finalforeach/cosmicreach/world/Chunk.java b/src/main/java/finalforeach/cosmicreach/world/Chunk.java
index a819628..1f3a55c 100644
--- a/src/main/java/finalforeach/cosmicreach/world/Chunk.java
+++ b/src/main/java/finalforeach/cosmicreach/world/Chunk.java
@@ -60,13 +60,12 @@ public class Chunk implements ISavedChunk<BlockState> {
          return 4096;
       } else {
          LayeredBlockData<BlockState> layered = (LayeredBlockData)numBlocksPerLayer;
-         int numBlocksPerLayer = 256;
          int max = 4096;
 
          for(int l = 15; l >= 0; --l) {
             IBlockLayer<BlockState> layer = layered.getLayer(l);
             if (layer instanceof BlockSingleLayer s) {
-               BlockState block = s.blockValue;
+               BlockState block = (BlockState) s.blockValue;
                if (!block.hasEmptyModel()) {
                   break;
                }
