diff --git a/src/main/java/finalforeach/cosmicreach/world/Region.java b/src/main/java/finalforeach/cosmicreach/world/Region.java
index 2e68be5..c04b252 100644
--- a/src/main/java/finalforeach/cosmicreach/world/Region.java
+++ b/src/main/java/finalforeach/cosmicreach/world/Region.java
@@ -5,6 +5,7 @@ import com.badlogic.gdx.utils.Array;
 import finalforeach.cosmicreach.blocks.BlockState;
 import finalforeach.cosmicreach.savelib.IRegion;
 
+@SuppressWarnings({"unused"})
 public class Region implements IRegion<BlockState> {
    public static final int REGION_WIDTH = 16;
    public static final int NUM_CHUNKS_IN_REGION = 4096;
