diff --git a/src/main/java/finalforeach/cosmicreach/world/Zone.java b/src/main/java/finalforeach/cosmicreach/world/Zone.java
index 809531f..e9f2067 100644
--- a/src/main/java/finalforeach/cosmicreach/world/Zone.java
+++ b/src/main/java/finalforeach/cosmicreach/world/Zone.java
@@ -20,16 +20,17 @@ import finalforeach.cosmicreach.worldgen.ZoneGenerator;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.util.Comparator;
 import java.util.Map;
 import java.util.PriorityQueue;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class Zone implements Json.Serializable, Disposable {
-   public PriorityQueue<ScheduledTrigger> eventQueue = new PriorityQueue<>((a, b) -> a.triggerTick() - b.triggerTick());
+   public PriorityQueue<ScheduledTrigger> eventQueue = new PriorityQueue<>(Comparator.comparingInt(ScheduledTrigger::triggerTick));
    public int currentTick = 0;
    private final Point3DMap<Chunk> chunks = new Point3DMap<>();
    private final Map<RegionCoords, Region> regionMap = new ConcurrentHashMap<>();
-   private transient Region[] regionValues = this.regionMap.values().toArray(new Region[this.regionMap.size()]);
+   private transient Region[] regionValues = this.regionMap.values().toArray(new Region[0]);
    public Array<Entity> allEntities = new Array<>();
    public Array<Player> players = new Array<>();
    public Vector3 spawnPoint;
@@ -125,7 +126,7 @@ public class Zone implements Json.Serializable, Disposable {
    public void addRegion(Region region) {
       synchronized(this.getRegionLock()) {
          this.regionMap.put(new RegionCoords(region.regionX, region.regionY, region.regionZ), region);
-         this.regionValues = this.regionMap.values().toArray(new Region[this.regionMap.size()]);
+         this.regionValues = this.regionMap.values().toArray(new Region[0]);
       }
    }
 
@@ -264,8 +265,7 @@ public class Zone implements Json.Serializable, Disposable {
       int cz = Math.floorDiv(z, 16);
       Chunk c = null;
 
-      for(int ci = 0; ci < candidateChunks.length; ++ci) {
-         Chunk candidateChunk = candidateChunks[ci];
+      for (Chunk candidateChunk : candidateChunks) {
          if (candidateChunk != null && candidateChunk.chunkX == cx && candidateChunk.chunkY == cy && candidateChunk.chunkZ == cz) {
             c = candidateChunk;
             break;
@@ -287,7 +287,7 @@ public class Zone implements Json.Serializable, Disposable {
    }
 
    public BlockState getBlockState(Chunk candidateChunk, int x, int y, int z) {
-      return this.get(candidateChunk, x, y, z, (c, lx, ly, lz) -> c.getBlockState(lx, ly, lz));
+      return this.get(candidateChunk, x, y, z, Chunk::getBlockState);
    }
 
    public BlockState getBlockState(Chunk candidateChunk, Chunk candidateChunkB, int x, int y, int z) {
@@ -395,7 +395,7 @@ public class Zone implements Json.Serializable, Disposable {
 
    @Override
    public void dispose() {
-      this.chunks.forEach(c -> c.dispose());
+      this.chunks.forEach(Chunk::dispose);
    }
 
    public void calculateSpawn() {
