diff --git a/src/main/java/finalforeach/cosmicreach/gamestates/PrealphaPreamble.java b/src/main/java/finalforeach/cosmicreach/gamestates/PrealphaPreamble.java
index 5406f08..f388ae8 100644
--- a/src/main/java/finalforeach/cosmicreach/gamestates/PrealphaPreamble.java
+++ b/src/main/java/finalforeach/cosmicreach/gamestates/PrealphaPreamble.java
@@ -71,11 +71,11 @@ public class PrealphaPreamble extends GameState {
 
                for(int w = 0; w < words.length; ++w) {
                   String word = words[w];
-                  float cw = curLine.length() == 0 ? 0.0F : lineDim.x;
+                  float cw = curLine.isEmpty() ? 0.0F : lineDim.x;
                   FontRenderer.getTextDimensions(this.uiViewport, word, lineDim);
                   lineDim.x += cw;
                   if (lineDim.x > maxW) {
-                     if (curLine.length() == 0) {
+                     if (curLine.isEmpty()) {
                         FontRenderer.drawText(batch, this.uiViewport, word, -8.0F, y, HorizontalAnchor.CENTERED, VerticalAnchor.CENTERED);
                      } else {
                         FontRenderer.drawText(batch, this.uiViewport, curLine, -8.0F, y, HorizontalAnchor.CENTERED, VerticalAnchor.CENTERED);
