diff --git a/src/main/java/finalforeach/cosmicreach/lwjgl3/CrashScreen.java b/src/main/java/finalforeach/cosmicreach/lwjgl3/CrashScreen.java
index a7b395a..b786672 100644
--- a/src/main/java/finalforeach/cosmicreach/lwjgl3/CrashScreen.java
+++ b/src/main/java/finalforeach/cosmicreach/lwjgl3/CrashScreen.java
@@ -55,6 +55,7 @@ public class CrashScreen {
       String gameVersion = "unknown";
 
       try {
+         System.out.println(new File("").getAbsolutePath());
          if (Gdx.files == null) {
             Gdx.files = new Lwjgl3Files();
          }
