diff --git a/src/main/java/finalforeach/cosmicreach/util/Point3DMap.java b/src/main/java/finalforeach/cosmicreach/util/Point3DMap.java
index c946e44..7a99a75 100644
--- a/src/main/java/finalforeach/cosmicreach/util/Point3DMap.java
+++ b/src/main/java/finalforeach/cosmicreach/util/Point3DMap.java
@@ -81,7 +81,7 @@ public class Point3DMap<T> implements IPoint3DMap<T> {
 
    @Override
    public void forEach(Consumer<T> consumer) {
-      this.map.forEach(lt -> ((IntMap)lt.value).forEach(t -> consumer.accept((T)t.value)));
+      this.map.forEach(lt -> lt.value.forEach(t -> consumer.accept((T)t.value)));
    }
 
    @Override
@@ -90,9 +90,9 @@ public class Point3DMap<T> implements IPoint3DMap<T> {
          long l = lt.key;
          int x = (int)(l >> 32);
          int y = (int)(l & 4294967295L);
-         ((IntMap)lt.value).forEach(it -> {
+         lt.value.forEach(it -> {
             int z = it.key;
-            consumer.consume((T)it.value, x, y, z);
+            consumer.consume(it.value, x, y, z);
          });
       });
    }
@@ -104,7 +104,7 @@ public class Point3DMap<T> implements IPoint3DMap<T> {
 
    @Override
    public void clear() {
-      this.map.forEach(m -> ((IntMap)m.value).clear());
+      this.map.forEach(m -> m.value.clear());
       this.map.clear();
    }
 }
